proctor-common/src/main/java/com/indeed/proctor/common/ProctorUtils.java

```diff
@@ -604,9 +605,9 @@ private static void verifyTest(
             }
 
             if (!unknownBuckets.isEmpty()) {
+                final String bucketString = (unknownBuckets.size() > 1 ? "bucket values: " : "bucket value: ") + Strings.join(unknownBuckets, ',');
                 throw new IncompatibleTestMatrixException(
-                        "Allocation range in " + testName + " from " + matrixSource
-                                + " refers to unknown bucket value(s) " + unknownBuckets + " with length > 0"
+                        "Application test specification does not contain "+ bucketString +". Please update the application test specification first"
```

Comment:
We usually call the specification "proctor specification" instead of "test specification". So how about "Proctor specification in your application does not ... Please update the proctor specification first"?

Reply:
Thanks @taitori . it is now updated.